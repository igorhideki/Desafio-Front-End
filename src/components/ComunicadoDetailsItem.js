import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import ComunicadoItemFile from './ComunicadoItemFile';

const handleCategoriaItem = (type) => {
  switch (type) {
    case 0:
      return { color: 'comunicados__item--orange', name: 'Informativos SUN' };
    case 1:
      return { color: 'comunicados__item--blue', name: 'Força de Vendas' };
    case 2:
      return { color: 'comunicados__item--green', name: 'Vivo Ligado' };
    case 3:
      return { color: 'comunicados__item--purple', name: 'Turbine Suas Vendas' };
    default:
      return { color: '', name: '' };
  }
};

const handleFormatDate = (date) => {
  const parseDate = moment(date, 'YYYY-MM-DD HH:mm');
  return moment(parseDate).format('DD/MM/YYYY - HH[h]mm');
};

const ComunicadoDetailsItem = (props) => (
  <div className={`comunicados__item comunicados__item--no-pointer ${handleCategoriaItem(props.type).color}`}>
    <div className="comunicados__row">
      <span className="comunicados__title">{props.title}</span>
      <span className="comunicados__id">{props.id}</span>
    </div>
    {props.files && props.files.length > 0 && props.files.map((file) => {
      return <ComunicadoItemFile key={file.name} {...file} />;
    })}
    <div className="comunicados__row">
      <span className="comunicados__type">{handleCategoriaItem(props.type).name}</span>
      <span className="comunicados__date">{handleFormatDate(props.date)}</span>
    </div>
    <div className="mt-m">
      {props.description !== '' && <p className="comunicados__subtitle">Resumo:</p>}
      <p className="comunicados__description">{props.description}</p>
      {props.description !== '' && <p className="comunicados__subtitle">Descrição:</p>}
      <p className="comunicados__description">{props.fullDescription}</p>
    </div>
  </div>
);

ComunicadoDetailsItem.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  date: PropTypes.string,
  description: PropTypes.string,
  fullDescription: PropTypes.string,
  type: PropTypes.number,
  files: PropTypes.arrayOf(PropTypes.shape({
    type: PropTypes.string,
    url: PropTypes.string,
    name: PropTypes.string
  }))
};

export default ComunicadoDetailsItem;
