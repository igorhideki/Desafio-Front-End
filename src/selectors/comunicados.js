import moment from 'moment';

const getVisiblesComunicados = (comunicados, { type = -1 }) => {
  const formatDate = 'YYYY-MM-DD HH:mm';
  return comunicados.filter((comunicado) => {
    if (type === -1) {
      return comunicado;
    }
    return comunicado.type === type;
  }).sort((a, b) => {
    return moment(a.date, formatDate).format() < moment(b.date, formatDate).format() ? 1 : -1;
  });
};

export default getVisiblesComunicados;

