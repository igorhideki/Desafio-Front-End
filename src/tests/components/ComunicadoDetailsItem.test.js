import React from 'react';
import { shallow } from 'enzyme';
import ComunicadoDetailsItem from '../../components/ComunicadoDetailsItem';
import comunicados from '../fixtures/comunicados';

test('should render ComunicadoDetailsItem', () => {
  const wrapper = shallow(<ComunicadoDetailsItem {...comunicados[2]} />);
  expect(wrapper).toMatchSnapshot();
});
