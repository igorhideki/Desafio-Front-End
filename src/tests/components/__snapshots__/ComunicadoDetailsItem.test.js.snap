// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ComunicadoDetailsItem 1`] = `
<div
  className="comunicados__item comunicados__item--no-pointer comunicados__item--blue"
>
  <div
    className="comunicados__row"
  >
    <span
      className="comunicados__title"
    >
      Novos Serviços
    </span>
    <span
      className="comunicados__id"
    >
      3
    </span>
  </div>
  <ComunicadoItemFile
    key="Material de divulgação"
    name="Material de divulgação"
    type="image/*"
    url="logob4.png"
  />
  <div
    className="comunicados__row"
  >
    <span
      className="comunicados__type"
    >
      Força de Vendas
    </span>
    <span
      className="comunicados__date"
    >
      02/05/2018 - 18h30
    </span>
  </div>
  <div
    className="mt-m"
  >
    <p
      className="comunicados__subtitle"
    >
      Resumo:
    </p>
    <p
      className="comunicados__description"
    >
      Novos serviços, clique para visualizar o material de divulgação.
    </p>
    <p
      className="comunicados__subtitle"
    >
      Descrição:
    </p>
    <p
      className="comunicados__description"
    >
      Novos serviços, clique para visualizar o material de divulgação.
    </p>
  </div>
</div>
`;
